src/controllers/facilitiesFeedbackController.js

```diff
@@ -0,0 +1,158 @@
+    if(error) return res.status(400).json({message: res.__(errorMessage)})
+    const currentUser = await decodeToken(req,res);
+    const {facility,facilityNotFound} = await findFacilityById(req.params.facilityId);
+    if(facilityNotFound == true)
+      return res.status(404).json({success: false, message: res.__("Facility not found")});
+     const newFeedback = {
+      facilityId:req.params.facilityId, senderId:currentUser.id, 
+      feedback_title: req.body.feedback_title, feedback_content: req.body.feedback_content, seen: false
+    }
+    let feedback;
+    try{
+        feedback = await  model.FacilitiesFeedback.create(newFeedback);
+    }catch(e){
+     return   res.status(500).json({sucess: false, message:res.___("Error")});
+    }
+    return res.status(201).json({
+        success: true,
+        message: res.__("Feedback sent successfully"),
+        feedback,
+        sender: currentUser,
+        facility
+    })
+}
+
+exports.getByFeedbackId = async(req,res) => {
+    const {feedback,feedbackNotFound} = await findFeedbackById(req.params.feedbackId);
+    if(feedbackNotFound)
+      return res.status(404).json({
+         success: false,
+         message: res.__("Feedback not found")
+      })
+    const {user, userNotFound} = await findUserById(feedback.dataValues.senderId);
+     if(userNotFound)
+      return res.status(400).json({message: res.__("Feedback sender not found")})
+     const {facility, notFound} = await findFacilityById(feedback.dataValues.facilityId);
+     if(notFound)  return res.status(400).json({message: res.__("Facility  not found")})
+
+     return res.status(200).json({
+         success:true,
+         feedback: {data: feedback,facility: facility.dataValues,sender: user.dataValues
+       }
+     })
+}
+
+exports.getAllFeedbacks = async(req,res) => {
+    const feedbacks = await model.FacilitiesFeedback.findAll();
+    return res.status(200).json({feedbacks});
+}
+
+exports.deleteFeedback = async(req,res) => {
+    const currentUser = await decodeToken(req,res);
+    const {feedback,feedbackNotFound} = await findFeedbackById(req.params.feedbackId);
+    if(feedbackNotFound)
+      return res.status(404).json({message: res.__("Feedback not found")})
+    if(currentUser.id != feedback.dataValues.senderId && currentUser.roleId != 3 && currentUser.roleId != 1)
+        return res.status(400).json({message:"Access denied!"})
+        try{
+          await model.FacilitiesFeedback.destroy({
+            where: { id: req.params.feedbackId }
+          });
+      }
+      catch(e){
+          return res.status(500).send({message:"Error"})
+      }
+      return res.status(200).json({
+          success: true,
+          message: res.__("Feedback deleted"),
+          feedback: feedback.dataValues
+      })
+}
+
+exports.updateFeedbackStatus = async(req,res) => {
+    const feedbackId = req.params.feedbackId;
+    const currentUser = await decodeToken(req);
```

Comment:
This line was repeat many times, you can create a middleware to deal with logged-in user.
Again, work on linting issues.

Reply:
Hello! actually that is a middleware function that returns the token details otherwise if the token is valid. I'm using it to be able to extract details within the token.